code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,360 @@
+					ubertarget.gib()
+			else
+				to_chat(ubertarget,"<span class='danger'>YOU ARE BULLETPROOF")
+				spawn(80)
+					animate(ubertarget)
+					ubertarget.color = null
+					ubertarget.flags &= ~INVULNERABLE
+					if(heartcheck)
+						heartcheck.damage += rand(5,60)
+		else
+			var/duration = 40
+
+			if(target_has_uberheart)
+				duration = 60
+
+			spawn(duration)
+				animate(ubertarget)
+				ubertarget.color = null
+				ubertarget.flags &= ~INVULNERABLE
+				if(heartcheck && !target_has_uberheart)
+					heartcheck.damage += rand(5,30)
+
+/obj/item/medigun/attackby(var/obj/item/weapon/W, var/mob/user)
+	if(istype(W, /obj/item/weapon/card/emag) && !emagged)
+		to_chat(user, "<span class='warning'>You swipe \the [W] through the circuits.</span>")
+		emagged = 1
+		medigunpack.emagged = 1
+		medigunpack.update_icon()
+		icon_state = "medigunred"
+		item_state = "medigunred"
+		user.update_inv_back()
+		user.update_inv_hands()
+		return
+	else
+		return ..()
+
+/obj/item/medigun/proc/heal(var/mob/living/M)
+	var/healing = 0.5
+	if(emagged)
+		healing = -1
+	if(istype(M,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/heart/heartcheck = H.internal_organs_by_name["heart"]
+		if(heartcheck.uber)
+			healing = 1
+
+	M.adjustOxyLoss(-2*healing)
+	M.heal_organ_damage(healing*2, 0)
+	M.heal_organ_damage(0, healing*2)
+	M.adjustToxLoss(-2*healing)
+	M.adjustCloneLoss(-2*healing)
+	M.adjustBrainLoss(-2*healing)
+	M.adjustBruteLoss(-2*healing)
+	M.adjustFireLoss(-2*healing)
+
+	if(healing > 0)
+		if(M.reagents.has_reagent("toxin"))
+			M.reagents.remove_reagent("toxin", healing)
+		if(M.reagents.has_reagent("stoxin"))
+			M.reagents.remove_reagent("stoxin", healing)
+		if(M.reagents.has_reagent("plasma"))
+			M.reagents.remove_reagent("plasma", healing)
+		if(M.reagents.has_reagent("sacid"))
+			M.reagents.remove_reagent("sacid", healing)
+		if(M.reagents.has_reagent("pacid"))
+			M.reagents.remove_reagent("pacid", healing)
+		if(M.reagents.has_reagent("cyanide"))
+			M.reagents.remove_reagent("cyanide", healing)
+		if(M.reagents.has_reagent("amatoxin"))
+			M.reagents.remove_reagent("amatoxin", healing)
+		if(M.reagents.has_reagent("chloralhydrate"))
+			M.reagents.remove_reagent("chloralhydrate", healing)
+		if(M.reagents.has_reagent("carpotoxin"))
+			M.reagents.remove_reagent("carpotoxin", healing)
+		if(M.reagents.has_reagent("mindbreaker"))
+			M.reagents.remove_reagent("mindbreaker", healing)
+
+		if(istype(M,/mob/living/simple_animal))
+			var/mob/living/simple_animal/SA = M
+			SA.health = min(SA.maxHealth,SA.health+2)
+
+		if(istype(M,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			for(var/datum/organ/external/temp in H.organs)
+				if(temp.status & ORGAN_BLEEDING)
+					temp.clamp()
+
+		if(M.losebreath >= 10)
+			M.losebreath = max(10, M.losebreath - 5)
+
+		if(M.bodytemperature > 310)
+			M.bodytemperature = max(310, M.bodytemperature - (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
```

Comment:
from setup.dm:

>//This is used in handle_temperature_damage() for humans, and in reagents that affect body temperature.

Think of the medigun's effects as the effects that a reagent would have on a mob (without there being actual reagents)